pkg/broker/broker.go

```diff
@@ -156,30 +160,47 @@ func (a AnsibleBroker) Provision(instanceUUID uuid.UUID, req *ProvisionRequest)
 		return nil, err
 	}
 
-	// TODO: Async? Bring in WorkEngine.
-	extCreds, err := apb.Provision(spec, parameters, a.clusterConfig, a.log)
-	if err != nil {
-		a.log.Error("broker::Provision error occurred.")
-		a.log.Error("%s", err.Error())
-		return nil, err
-	}
+	var token string
+
+	if async {
+		a.log.Info("ASYNC provisioning in progress")
+		// asyncronously provision and return the token for the lastoperation
+		pjob := NewProvisionJob(spec, parameters, a.clusterConfig, a.log)
+
+		// HACK: wow this feels dirty
+		a.engine.AttachSubscriber(NewProvisionWorkSubscriber(a.dao))
```

Comment:
It was something we used in our cap-server prototype, so we brought it over from there since it worked ok. Our initial use was to get current updates to a websocket on the UI. I think ultimately that would be something we'd desire is to get more fine grained updates that we can write output to.